compositor_pipeline/src/pipeline/decoder.rs

```diff
@@ -1,65 +1,19 @@
-use crate::{audio_mixer::InputSamples, error::DecoderInitError, queue::PipelineEvent};
+use crate::{audio_mixer::InputSamples, queue::PipelineEvent};
 
-use super::{input::ChunksReceiver, structs::VideoCodec};
+use super::structs::VideoCodec;
 
 use bytes::Bytes;
-use compositor_render::{Frame, InputId};
-use crossbeam_channel::{bounded, Receiver};
+use compositor_render::Frame;
+use crossbeam_channel::Receiver;
 
 pub use audio::AacDecoderError;
 
 mod audio;
 mod video;
 
-pub fn start_decoder(
-    input_id: InputId,
```

Comment:
technically this code could still be here, but given that it would also need to handle a case without decoder it seems more appropriate to move it to the input